cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/similarity/Similarity.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.similarity
+
+import com.twitter.cassovary.graph.{DirectedGraph, Node}
+import com.twitter.cassovary.graph.GraphDir.GraphDir
+import scala.collection.mutable
+
+trait Similarity {
+
+  def graph: DirectedGraph[Node]
+
+  protected def getNeighbors(dir: GraphDir, nodeId: Int) = graph.getNodeById(nodeId).get.neighborIds(dir).toSet
```

Comment:
what should the similarity score between nodeIds that exists and doesn't exist be? zero? negative number?